マルチスレッド環境で使うとcom.sun.star.lang.IllegalArgumentExceptionが発生する

### 概要

Webサーバーなどのマルチスレッド環境では一時ファイルのファイルパスが重複する場合があり、com.sun.star.lang.IllegalArgumentExceptionが発生する。

```
Caused by: org.artofsolving.jodconverter.office.OfficeException: could not load document: 1468386966787.xls
    at org.artofsolving.jodconverter.AbstractConversionTask.loadDocument(AbstractConversionTask.java:92)
    at org.artofsolving.jodconverter.AbstractConversionTask.execute(AbstractConversionTask.java:59)
    at org.artofsolving.jodconverter.office.ExternalOfficeManager.execute(ExternalOfficeManager.java:80)
    at org.artofsolving.jodconverter.OfficeDocumentConverter.convert(OfficeDocumentConverter.java:78)
    at org.artofsolving.jodconverter.OfficeDocumentConverter.convert(OfficeDocumentConverter.java:69)
    at org.bbreak.excella.reports.exporter.OoPdfExporter.output(OoPdfExporter.java:160)
    ... 46 more
Caused by: com.sun.star.lang.IllegalArgumentException: Unsupported URL <file:///C:/Users/USER1/AppData/Local/Temp/1468386966787.xls>: "type detection failed" 
    at com.sun.star.lib.uno.environments.remote.Job.remoteUnoRequestRaisedException(Job.java:182)
    at com.sun.star.lib.uno.environments.remote.Job.execute(Job.java:148)
    at com.sun.star.lib.uno.environments.remote.JobQueue.enter(JobQueue.java:344)
    at com.sun.star.lib.uno.environments.remote.JobQueue.enter(JobQueue.java:313)
    at com.sun.star.lib.uno.environments.remote.JavaThreadPool.enter(JavaThreadPool.java:101)
    at com.sun.star.lib.uno.bridges.java_remote.java_remote_bridge.sendRequest(java_remote_bridge.java:652)
    at com.sun.star.lib.uno.bridges.java_remote.ProxyFactory$Handler.request(ProxyFactory.java:154)
    at com.sun.star.lib.uno.bridges.java_remote.ProxyFactory$Handler.invoke(ProxyFactory.java:136)
    at com.sun.proxy.$Proxy61.loadComponentFromURL(Unknown Source)
    at org.artofsolving.jodconverter.AbstractConversionTask.loadDocument(AbstractConversionTask.java:90)
    ... 51 more
```

ExternalOfficeManagerはJavadocに記載されているとおり、Officeとの通信処理は`serializable`ブロック内に実装されているため、一時ファイルのファイルパスが重複すると最初にPDF変換されるファイルのみ変換に成功し、以降のファイルは一時ファイルを開くことが出来ないため変換に失敗する。

```
// ExternalOfficeManagerのJavadocより
This OfficeManager implementation basically provides the same behaviour as JODConverter 2.x, including using synchronized blocks for serialising office operations.
```

--- a/src/main/java/org/bbreak/excella/reports/exporter/OoPdfExporter.java
+++ b/src/main/java/org/bbreak/excella/reports/exporter/OoPdfExporter.java
@@ -151,7 +151,8 @@ public class OoPdfExporter extends ReportBookExporter {
 
             // 一時フォルダに吐き出し
             ExcelExporter excelExporter = new ExcelExporter();
-            String tmpFileName = System.getProperty( "java.io.tmpdir") + System.currentTimeMillis();
+            tmpFile = File.createTempFile( getClass().getSimpleName(), null);
+            String tmpFileName = tmpFile.getCanonicalPath();
             excelExporter.setFilePath( tmpFileName);
             excelExporter.output( book, bookdata, null);
 

